ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java

```diff
@@ -804,19 +843,42 @@ void handleBody(InputStream payload)
               MetadataContentSerDe.deserializeMetadataContentRecord(serializedMetadataContent, blobIdFactory);
           chunkSize = compositeBlobInfo.getChunkSize();
           totalSize = compositeBlobInfo.getTotalSize();
-          List<StoreKey> keys = compositeBlobInfo.getKeys();
-          chunkIdIterator = keys.listIterator();
-          numChunksTotal = keys.size();
-          dataChunks = new GetChunk[Math.min(keys.size(), NonBlockingRouter.MAX_IN_MEM_CHUNKS)];
-          for (int i = 0; i < dataChunks.length; i++) {
-            dataChunks[i] = new GetChunk(chunkIdIterator.nextIndex(), (BlobId) chunkIdIterator.next());
+          if (hasRange && (chunkSize == CompositeBlobInfo.UNDEFINED_CHUNK_SIZE
```

Comment:
Let us discuss this tomorrow - I think we can simply get rid of Metadata content V1 and simplify `CompositeBlobInfo`. That would make the code leaner.